src/graph_export/graph_cli_parser.cpp

```diff
@@ -0,0 +1,158 @@
+                cxxopts::value<std::string>()->default_value("10"),
+                "CACHE_SIZE")
+            ("draft_source_model",
+                "HF model name or path to the local folder with PyTorch or OpenVINO draft model.",
+                cxxopts::value<std::string>()->default_value(""),
+                "DRAFT_SOURCE_MODEL")
+            ("dynamic_split_fuse",
+                "Dynamic split fuse algorithm enabled. Default true.",
+                cxxopts::value<std::string>()->default_value("false"),
+                "DYNAMIC_SPLIT_FUSE");
+
+        options->add_options("plugin config")
+            ("max_prompt_len",
+                "Sets NPU specific property for maximum number of tokens in the prompt.",
+                cxxopts::value<std::string>()->default_value(""),
+                "MAX_PROMPT_LEN")
+            ("kv_cache_precision",
+                "u8 or empty (model default). Reduced kv cache precision to u8 lowers the cache size consumption.",
+                cxxopts::value<std::string>()->default_value(""),
+                "KV_CACHE_PRECISION");
+}
+
+void GraphCLIParser::printHelp() {
+    if (!this->options) {
+        this->createOptions();
+    }
+    std::cout << options->help({"general options", "plugin config"}) << std::endl;
+}
+
+void GraphCLIParser::parse(const std::vector<std::string>& unmatchedOptions) {
+    try {
+        if (!this->options) {
+            this->createOptions();
+        }
+        std::vector<const char*> cStrArray;
+        cStrArray.reserve(unmatchedOptions.size()+1);
+        cStrArray.push_back("ovms graph");
+        for (const auto& str : unmatchedOptions) {
+            cStrArray.push_back(str.c_str());
+        }
+        const char* const* args = cStrArray.data();
+        result = std::make_unique<cxxopts::ParseResult>(options->parse(cStrArray.size(), args));
+
+        if (result->unmatched().size()) {
+            std::cerr << "error parsing options - unmatched arguments: ";
+            for (auto& argument : result->unmatched()) {
+                std::cerr << argument << ", ";
+            }
+            std::cerr << std::endl;
+            exit(OVMS_EX_USAGE);
+        }
+    } catch (const std::exception& e) {
+        std::cerr << "error parsing options: " << e.what() << std::endl;
+        exit(OVMS_EX_USAGE);
+    }
+}
+
+void GraphCLIParser::prepare(ServerSettingsImpl* serverSettings, ModelsSettingsImpl* modelsSettings) {
+    if (nullptr == result) {
+        // Pull with default arguments - no arguments from user
+        if (serverSettings->hfSettings.pullHfModelMode) {
+            serverSettings->hfSettings.graphSettings = GraphCLIParser::defaultGraphSettings();
+            return;
+        } else {
+            throw std::logic_error("Tried to prepare server and model settings without graph parse result");
+        }
+    }
+
+    serverSettings->hfSettings.graphSettings.maxNumSeqs = result->operator[]("max_num_seqs").as<std::string>();
```

Comment:
Nitpick but wouldn't it be easier with:
`auto& graphSettings = hfSettings.graphSettings;
graphSettings.pipelineType = result->operator[]("pipeline_type").as<std::string>();`